docs/component-documentation.md

```diff
@@ -0,0 +1,73 @@
+# Component Documentation Best Practices
+
+This guide outlines the best practices for documenting components in the MetaMask Design System.
+
+## Documentation Structure
+
+Each component should have a README file that serves as the comprehensive documentation:
+
+- For React components: `packages/design-system-react/src/components/ComponentName/README.mdx`
+- For React Native components: `packages/design-system-react-native/src/components/ComponentName/README.md`
+
+## Example Templates
+
+We provide example templates for both platforms that demonstrate the proper documentation structure:
+
+- [React Component Documentation Example](./component-readme-examples/react-readme-example.mdx)
+- [React Native Component Documentation Example](./component-readme-examples/react-native-readme-example.md)
+
+### Required Sections
+
+1. **Component Overview**
+
+   - Component name
+   - Brief description of the component's purpose
+   - Basic usage example with imports
+
+2. **Props Documentation**
+   It should then describe each prop and an example of it's usage
+   - `propName`
+   - Brief description of the prop and its usage
+   - Any enums that are used with that prop
+   - Prop type table documenting type, required, defaults
+   - Story or code example
```

Comment:
One of the main differences between MD and MDX is that we are going to use the storybook `<Canvas />` for both the preview of the component and the code example using the show code button


https://github.com/user-attachments/assets/4bf96e29-484a-4efa-a1f5-e0e4279a7a2a